net/pb/net.proto

```diff
@@ -42,10 +34,51 @@ message PushLogRequest {
         // creator is the PeerID of the peer that created the log.
         string creator = 4;
         // log hold the block that represent version of the document.
-        Document.Log log = 6;
+        Log log = 6;
     }
 }
 
+// EncryptionKeyTarget is a struct containing information about required doc encryption key.
+message EncryptionKeyTarget {
+    // docID is the ID of the document that the key is being requested for.
+    bytes docID = 1;
+    // fieldName if the name of the document field that the key is being requested for.
+    // If the fieldName is empty, the key is being requested for the whole document.
+    string fieldName = 2;
+    // keyID is the hash (Cid) of the key that is being requested.
+    bytes keyID = 3;
+}
+
+// FetchEncryptionKeyRequest is a request to receive a doc encryption key
+// from a peer that holds it.
+message FetchEncryptionKeyRequest {
+    // targets is the list of docs/fields for which encryption keys are being requested.
+    repeated EncryptionKeyTarget targets = 1;
+    // schemaRoot is the SchemaRoot of the collection that the document resides in.
+    bytes schemaRoot = 2;
```

Comment:
it's included in the request so that we can receive it back in the reply. After we received enc key we send an event that included schema root and the enc key. The merger handling the event can assosiate the doc with the collection based on the schema root. Otherwise it doesn't know which collection it belongs to.
But I think we can include schema root in a session the same way we do with ephemeral keys. Will this be fine? Or you want to propose another way of linking a doc to a collection?

Reply:
We can leave it as is. 👍